app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance = self._get_first_group_instance_in_block(group, blocks)
+            if first_group_instance is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance, group['blocks'][0]['id'])
 
-            if blocks:
-                path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
+                if blocks:
```

Comment:
Doesn't this still need to be called regardless of whether we have a `first_group_instance`?

Reply:
I dont think so  ,what it is trying to do is to not add the path for those blocks unless there is at least one block that is not skipped